prboom2/data/lumps/gls_main.lmp

```diff
@@ -1,3 +1,15 @@
+#version 110
+
+#ifdef VERTEX
```

Comment:
Why do we want two files squeezed into one like this?

Reply:
Putting all shader stages in the same file is typical in HLSL, [Slang](https://github.com/shader-slang/slang/blob/master/examples/triangle/shaders.slang), [WGSL](https://github.com/gfx-rs/wgpu-native/blob/trunk/examples/triangle/shader.wgsl), [Metal](https://metalbyexample.com/up-and-running-2/), etc.  Since the stages are intimately linked, you can keep everything in one place, specify `varying`s only once, etc.  It's like not having to have separate `.c`/`.h` files.

Reply:
Understood 👍